cpp/src/arrow/type.h

```diff
@@ -403,19 +403,6 @@ class ARROW_EXPORT ListType : public NestedType {
   std::string name() const override { return "list"; }
 };
 
-namespace meta {
-
-/// Additional ListType class that can be instantiated with only compile-time arguments.
-template <typename T>
-class ARROW_EXPORT ListType : public ::arrow::ListType {
```

Comment:
@xhochy This class failed linking from `arrow.dll`. Not sure why...

Reply:
Hm, should open a follow up issue to investigate

Reply:
I don't need the type for the current functionality but I'm working very slowly on adding some more interfaces that allow construction of Arrow classes using meta programming. There this type will be very helpful. It's ok for me to drop it temporarily but I may readd a similar functionality again some time.

Reply:
I added recently a new macro `ARROW_TEMPLATE_EXPORT` that solves this, see https://github.com/apache/arrow/blob/master/cpp/src/arrow/util/visibility.h#L56. Let me see if I can add this type back in

Reply:
I'm not sure how it would, but I'll let you give it a try ;-)

Reply:
Eh, I won't rock the boat. Let's look into it another time